Runner/Program.cs

```diff
@@ -184,29 +209,21 @@ private static ServiceProvider BuildDependencyInjection(IConfiguration config) =
                 .AddTransient<ValidationClient>()
                 .AddSingleton(provider =>
                 {
-                    var rules = new IValidationRule[]
+                    var logger = provider.GetService<ILogger<Program>>();
+                    var rules = selectedValidationRules.Select(r => (IValidationRule)provider.GetService(r)).ToArray();
```

Comment:
I think this part doesn't work. but anyway, I think we can have a better implementation for this.

I think we should create an extension method for `ServiceCollection` for which we just pass the rule config as a parameter. This method should be in `ValidationLibrary.Rules` so we can make that Assembly loading part less error prone.

Usage:
```
provider.AddValidationRules(config)
```

and in `RepositoryValidator` creation we can just `provider.GetServices<IValidationRule>()` which should return all implementations.


Then we can also use this in `ValidationLibrary.AzureFunctions.Startup`

Reply:
I think this is good and this has been changed as suggested except for the "provider.GetServices<IValidationRule>()" part which doesn't seem to work. I have been working on that part.